### PR TITLE
AUTH-26: CSR Approver manifests

### PR DESCRIPTION
**Pull request by slaskawi**
https://issues.redhat.com/browse/AUTH-26

This roles are required for https://github.com/openshift/cluster-policy-controller/pull/61

### CODE REVIEW

File: manifests/0000_30_cluster_csr_approver_02_clusterrolebinding.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+kind: ClusterRoleBinding
+metadata:
+  name: system:openshift:operator:cluster-csr-approver-controller
+  annotations:
+    include.release.openshift.io/ibm-cloud-managed: "true"
+    include.release.openshift.io/self-managed-high-availability: "true"
+    include.release.openshift.io/single-node-developer: "true"
+roleRef:
+  kind: ClusterRole
+  name: cluster-admin
```
Comment (deads2k) [inline manifests/0000_30_cluster_csr_approver_02_clusterrolebinding.yaml:11]: this doesn't look right.  I expected to see a clusterrole that has the correct permissions.

**Comment by deads2k** [comment]
these appear to be for the operand.  You want to create these in bindata/v4 something and then add them to the staticresourcecontroller.

Having our operator manage the resources for our operand gives us much greater flexibility in the future.

File: manifests/0000_30_cluster_csr_approver_02_clusterrolebinding.yaml (hunk 1 above)
Comment (stlaz) [reply manifests/0000_30_cluster_csr_approver_02_clusterrolebinding.yaml:11]: +1, this has to be refined

File: manifests/0000_30_cluster_csr_approver_01_serviceaccount.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  namespace: openshift-kube-controller-manager
+  name: cluster-csr-approver-controller
```
Comment (stlaz) [inline manifests/0000_30_cluster_csr_approver_01_serviceaccount.yaml:5]: what is this, is this used anywhere?

Comment (stlaz) [reply manifests/0000_30_cluster_csr_approver_01_serviceaccount.yaml:5]: oh I see, the client builder in CPC uses the `SAControllerClientBuilder` that allows using different SAs per controller. TIL.

File: manifests/0000_30_cluster_csr_approver_02_clusterrolebinding.yaml (hunk 1 above)
Comment (slaskawi) [reply manifests/0000_30_cluster_csr_approver_02_clusterrolebinding.yaml]: Fixed. Thanks for pointing this out!

File: manifests/0000_30_cluster_csr_approver_01_serviceaccount.yaml (hunk 2 above)
Comment (slaskawi) [reply manifests/0000_30_cluster_csr_approver_01_serviceaccount.yaml]: Resolving this comment as I moved the definitions along with @deads2k's suggestion.

**Comment by slaskawi** [comment]
> these appear to be for the operand. You want to create these in bindata/v4 something and then add them to the staticresourcecontroller.

@deads2k Thanks for the suggestion. Fixed.

@deads2k @stlaz @s-urbaniak Updated according to the comments and ready for the next round of review.

File: bindata/v4.1.0/kube-controller-manager/csr_approver_serviceaccount.yaml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  namespace: openshift-kube-controller-manager
+  name: cluster-csr-approver-controller
```
Comment (deads2k) [inline bindata/v4.1.0/kube-controller-manager/csr_approver_serviceaccount.yaml:5]: I'm a little surprised this is required instead of auto-created, but ok.

Comment (slaskawi) [reply bindata/v4.1.0/kube-controller-manager/csr_approver_serviceaccount.yaml:5]: hmmmm just to educate me - would you mind to explain how this auto-generate mechanism works and where I can find it? Maybe I implement this somehow differently?

**Comment by stlaz** [comment]
/hold
no JIRA tickets in commit messages

File: bindata/v4.1.0/kube-controller-manager/csr_approver_serviceaccount.yaml (hunk 3 above)
Comment (stlaz) [reply bindata/v4.1.0/kube-controller-manager/csr_approver_serviceaccount.yaml:5]: I think the ClientBuilder you're using in the CPO PR is capable of creating the SA

File: pkg/operator/v411_00_assets/bindata.go [hunk 4]
```diff
@@ -4,6 +4,9 @@
 // bindata/v4.1.0/config/defaultconfig.yaml
 // bindata/v4.1.0/kube-controller-manager/cluster-policy-controller-cm.yaml
 // bindata/v4.1.0/kube-controller-manager/cm.yaml
+// bindata/v4.1.0/kube-controller-manager/csr_approver_clusterrole.yaml
```
Comment (stlaz) [inline pkg/operator/v411_00_assets/bindata.go:7]: move autogenerated changes to a separate commit, please

File: bindata/v4.1.0/kube-controller-manager/csr_approver_serviceaccount.yaml (hunk 3 above)
Comment (slaskawi) [reply bindata/v4.1.0/kube-controller-manager/csr_approver_serviceaccount.yaml:5]: Let me verify that by creating a new cluster without this commit. Will let you know how did it go.

**Comment by slaskawi** [comment]
> no JIRA tickets in commit messages

@stlaz Ok, I can remove it. But perhaps you'd like to reconsider? Adding JIRA IDs into the commits automatically adds Pull Request references to the ticket. This is the simplest way you can maintain a mapping between JIRAs and the code. See this screenshot for the reference:

image

File: pkg/operator/v411_00_assets/bindata.go (hunk 4 above)
Comment (slaskawi) [reply pkg/operator/v411_00_assets/bindata.go:7]: Done.

File: bindata/v4.1.0/kube-controller-manager/csr_approver_serviceaccount.yaml (hunk 3 above)
Comment (slaskawi) [reply bindata/v4.1.0/kube-controller-manager/csr_approver_serviceaccount.yaml:5]: Yes, you're absolutely correct! Although the Service Account is created in `openshift-infra` namespace, which is quite surprising (it thought it would be created in `openshift-kube-controller-manager`, where the controller runs).

Closing this PR.

**Comment by slaskawi** [comment]
Closing this PR as the SA is automatically created, see https://github.com/openshift/cluster-kube-controller-manager-operator/pull/535#discussion_r652096644

**Comment by slaskawi** [comment]
> Closing this PR as the SA is automatically created, see #535 (comment)

@stlaz pointed out that I miss the RBAC part :facepalm: Reopening.

**Comment by slaskawi** [comment]
@deads2k @stlaz @s-urbaniak Updated, rebased and ready for the next round of reviews.

**Comment by stlaz** [comment]
/hold cancel
> But perhaps you'd like to reconsider? Adding JIRA IDs into the commits automatically adds Pull Request references to the ticket.

Having it in the PR name should be enough. If the tooling is incapable to do that, it's bad tooling.
